### PR TITLE
ObjectPool notes about IDisposable support

### LINKED ISSUE #16747: Incorrect statement about IDisposable support
This statement is no longer correct:

WARNING: The ObjectPool doesn't implement IDisposable. We don't recommend using it with types that need disposal.

Please update the documentation to reflect recent changes. See: 

https://github.com/dotnet/extensions/issues/973

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 075aad4c-9e53-92c1-8405-70bd5ed54baa
* Version Independent ID: fa535bff-449d-ac12-ea5b-8a975b0682f6
* Content: [Object reuse with ObjectPool in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/performance/objectpool?view=aspnetcore-3.1#feedback)
* Content Source: [aspnetcore/performance/ObjectPool.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/performance/ObjectPool.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-performance**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by gfoidl**
Fixes #16747

/cc: @Rick-Anderson

### CODE REVIEW

**Comment by gfoidl** [comment]
The change (https://github.com/dotnet/extensions/pull/977) is available in ASP.NET Core 3.0 onwards.
How can I specify the version in the md-file?

File: aspnetcore/performance/ObjectPool.md [hunk 1]
```diff
@@ -50,7 +48,14 @@ The ObjectPool can be used in an app in multiple ways:
 
 ## How to use ObjectPool
 
-Call <xref:Microsoft.Extensions.ObjectPool.ObjectPool`1> to get an object and <xref:Microsoft.Extensions.ObjectPool.ObjectPool`1.Return*> to return the object.  There's no requirement that you return every object. If you don't return an object, it will be garbage collected.
+Call <xref:Microsoft.Extensions.ObjectPool.ObjectPool`1.Get*> to get an object and <xref:Microsoft.Extensions.ObjectPool.ObjectPool`1.Return*> to return the object.  There's no requirement that you return every object. If you don't return an object, it will be garbage collected.
```
Comment (gfoidl) [inline aspnetcore/performance/ObjectPool.md:51]: Just specified `Get` as method here. Before it pointed to the pool-class.

Comment (Rick-Anderson) [reply aspnetcore/performance/ObjectPool.md:51]:
```
::: moniker range=">= aspnetcore-3.0"

// 3.0 and higher
::: moniker-end

::: moniker range="< aspnetcore-3.0"


::: moniker-end
```

**Comment by Rick-Anderson** [comment]
@gfoidl  much appreciated. I put you on the authors list. I did some minor editing. Great works.

**Comment by gfoidl** [comment]
@Rick-Anderson thanks!

**Pull request merged**